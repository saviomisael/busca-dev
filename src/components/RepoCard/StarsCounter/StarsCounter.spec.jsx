import { screen } from '@testing-library/react';
import { StarsCounter } from '.';
import StarEmpty from '../../../assets/images/star_empty.svg';
import StarFilled from '../../../assets/images/star_fill.svg';

describe('<StarsCounter />', () => {
  it('should renders component with starsCount into content', () => {
    global.renderTheme(<StarsCounter starsCount={5} />);

    const counter = screen.getByText(/estrelas/i);

    expect(counter.textContent).toContain(5);
  });

  it('should renders star empty icon when starsCount is equal to zero', () => {
    global.renderTheme(<StarsCounter starsCount={0} />);

    const icon = screen.getByAltText('star icon');

    expect(icon.src).toContain(StarEmpty);
  });

  it('should renders star filled icon when starsCount greater than zero', () => {
    global.renderTheme(<StarsCounter starsCount={5} />);

    const icon = screen.getByAltText('star icon');

    expect(icon.src).toContain(StarFilled);
  });
});
