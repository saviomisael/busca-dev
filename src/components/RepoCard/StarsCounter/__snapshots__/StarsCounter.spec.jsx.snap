// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarsCounter /> should matches with snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.7rem;
  column-gap: 0.7rem;
}

.c2 {
  text-transform: uppercase;
  font-size: 1.6rem;
  color: #284B63;
}

.c1 {
  -webkit-filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
  filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
}

<div
  class="c0"
>
  <img
    alt="star icon"
    class="c1"
    src="star_fill.svg"
  />
  <p
    class="c2"
  >
    5
     Estrelas
  </p>
</div>
`;
